packages/esm-patient-chart-app/src/visit/visit-form/visit-form.component.tsx

```diff
@@ -817,52 +818,47 @@ interface VisitFormExtensionSlotProps {
   name: string;
   patientUuid: string;
   visitFormOpenedFrom: string;
-  setOnVisitCreatedOrUpdatedCallbacks: React.Dispatch<
-    React.SetStateAction<Map<string, OnVisitCreatedOrUpdatedCallback>>
-  >;
+  setVisitFormCallbacks: React.Dispatch<React.SetStateAction<Map<string, VisitFormCallbacks>>>;
 }
 
 type VisitFormExtensionState = {
   patientUuid: string;
 
   /**
-   * This function allows an extension to register a callback to run after a visit has been created.
-   * This callback can be used to make further requests. The callback should handle its own UI notification
+   * This function allows an extension to register callbacks for visit form submission.
+   * This callbacks can be used to make further requests. The callbacks should handle its own UI notification
    * on success / failure, and its returned Promise MUST resolve on success and MUST reject on failure.
    * @param callback
    * @returns
    */
-  setOnVisitCreatedOrUpdated: (callback: OnVisitCreatedOrUpdatedCallback) => void;
+  setVisitFormCallbacks(callbacks: VisitFormCallbacks);
 
   visitFormOpenedFrom: string;
   patientChartConfig: ChartConfig;
 };
 
-const VisitFormExtensionSlot: React.FC<VisitFormExtensionSlotProps> = ({
-  name,
-  patientUuid,
-  visitFormOpenedFrom,
-  setOnVisitCreatedOrUpdatedCallbacks,
-}) => {
-  const config = useConfig<ChartConfig>();
-
-  return (
-    <ExtensionSlot name={name}>
-      {(extension) => {
-        const state: VisitFormExtensionState = {
-          patientUuid,
-          setOnVisitCreatedOrUpdated: (callback) => {
-            setOnVisitCreatedOrUpdatedCallbacks((old) => {
-              return new Map(old).set(extension.id, callback);
-            });
-          },
-          visitFormOpenedFrom,
-          patientChartConfig: config,
-        };
-        return <Extension state={state} />;
-      }}
-    </ExtensionSlot>
-  );
-};
+const VisitFormExtensionSlot: React.FC<VisitFormExtensionSlotProps> = React.memo(
+  ({ name, patientUuid, visitFormOpenedFrom, setVisitFormCallbacks }) => {
+    const config = useConfig<ChartConfig>();
+
+    return (
+      <ExtensionSlot name={name}>
+        {(extension: AssignedExtension) => {
+          const state: VisitFormExtensionState = {
+            patientUuid,
+            setVisitFormCallbacks: (callbacks) => {
+              setVisitFormCallbacks((old) => {
+                return new Map(old).set(extension.id, callbacks);
```

Comment:
Do we need a new `Map` on every call? It might simplify things if it's possible to do an update-in-place.

Reply:
I think we do, if we keep using `useState`, as the Map should be immutable. (This is akin of editing / adding an element to an array state). That said, I think we should move to `useRef`.